price/src/price_generator.rs

```diff
@@ -11,13 +11,13 @@ use tokio::time;
 
 #[derive(Debug, Clone, Serialize)]
 pub struct Price {
-    pub timestamp: i64,
+    pub timestamp: DateTime<Utc>,
     pub price: u64,
     pub token_type: BlockchainTokenTypeV1,
 }
 
 impl Price {
-    pub fn new(timestamp: i64, price: u64, token_type: BlockchainTokenTypeV1) -> Self {
+    pub fn new(timestamp: DateTime<Utc>, price: u64, token_type: BlockchainTokenTypeV1) -> Self {
```

Comment:
I'd recommend making this new function `pub(crate)`. I don't think there's ever a need to construct this outside of this crate

Reply:
I just removed pub and everything compiled 29f472a